### PR TITLE
Enable MPI on CircleCI

### LINKED ISSUE #2: Enable MPI tests in CI
We currently don't test MPI functionality anywhere. It needs to be added to at least one CI configuration.

### PR DESCRIPTION
**Pull request by aurianer**
Fixes #2.

### CODE REVIEW

**Comment by msimberg** [comment]
bors try

File: .github/workflows/linux_hip.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
   build:
     name: github/linux/hip/fast
     runs-on: ubuntu-latest
-    container: pikaorg/pika-ci-hip:2
+    container: pikaorg/pika-ci-hip:3
```
Comment (msimberg) [inline .github/workflows/linux_hip.yml:23]:
```suggestion
    container: pikaorg/pika-ci-hip:2
```

**Comment by aurianer** [comment]
bors try

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
bors try

**Review by msimberg — APPROVED** [review]

**Comment by msimberg** [comment]
bors merge

**Pull request merged**